### PR TITLE
Add stateful_ips to region_per_instance_config and per_instance_config

### PR DESCRIPTION
**Pull request by hao-nan-li**
Add stateful_ips to region_per_instance_config and per_instance_config

https://cloud.google.com/compute/docs/reference/rest/beta/instanceGroupManagers/createInstances

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added `internal_ip` and `external_ip` to `google_compute_per_instance_config` and `google_compute_region_per_instance_config` (beta)
```

### CODE REVIEW

**Comment by ScottSuarez** [comment]
rerolling since I'm going to oof tmrw and tuesday

File: mmv1/products/compute/api.yaml [hunk 1]
```diff
@@ -9750,6 +9750,70 @@ objects:
...
+              Preserved internal IPs defined for this instance. This map is keyed with the name of the network interface.
+            value_type: !ruby/object:Api::Type::NestedObject
+              properties:
+                - !ruby/object:Api::Type::Enum
+                  name: autoDelete
+                  description: |
+                    A value that prescribes what should happen to the stateful disk when the VM instance is deleted.
+                    The available options are `NEVER` and `ON_PERMANENT_INSTANCE_DELETION`.
+                    `NEVER` - detach the disk when the VM is deleted, but do not delete the disk.
+                    `ON_PERMANENT_INSTANCE_DELETION` will delete the stateful disk when the VM is permanently
```
Comment (askubis) [inline mmv1/products/compute/api.yaml:9768]: Address

File: mmv1/products/compute/api.yaml [hunk 2]
```diff
@@ -9750,6 +9750,70 @@ objects:
...
+            min_version: beta
+            key_name: "interface_name"
+            description: |
+              Preserved internal IPs defined for this instance. This map is keyed with the name of the network interface.
+            value_type: !ruby/object:Api::Type::NestedObject
+              properties:
+                - !ruby/object:Api::Type::Enum
+                  name: autoDelete
+                  description: |
+                    A value that prescribes what should happen to the stateful disk when the VM instance is deleted.
```
Comment (askubis) [inline mmv1/products/compute/api.yaml:9765]: address

Comment (askubis) [reply mmv1/products/compute/api.yaml:9765]: here, and generally throughout the comments

File: mmv1/products/compute/api.yaml (hunk 1 above)
Comment (hao-nan-li) [reply mmv1/products/compute/api.yaml:9768]: Thanks for pointing it out, I'll change the description to what's on the public API page.
https://cloud.google.com/compute/docs/reference/rest/beta/instanceGroupManagers/patchPerInstanceConfigs

File: mmv1/products/compute/api.yaml [hunk 3]
```diff
@@ -9750,6 +9750,62 @@ objects:
                     - :NEVER
                     - :ON_PERMANENT_INSTANCE_DELETION
                   default_value: :NEVER
+          - !ruby/object:Api::Type::Map
+            name: 'internalIp'
+            api_name: internalIPs
+            min_version: beta
+            key_name: "interface_name"
```
Comment (shuyama1) [inline mmv1/products/compute/api.yaml:9757]: Do we want to add `key_description` here which will add a description for the field `interface_name`  in doc? Current description is "`interface_name` - (Required) The identifier for this object. Format specified above". I wonder if we want to change it.

File: mmv1/products/compute/terraform.yaml [hunk 4]
```diff
@@ -1961,6 +1961,10 @@ overrides: !ruby/object:Overrides::ResourceOverrides
         is_set: true
         custom_flatten: templates/terraform/custom_flatten/preserved_state_disks.go.erb
         custom_expand: templates/terraform/custom_expand/preserved_state_disks.go.erb
+      preservedState.internalIp: !ruby/object:Overrides::Terraform::PropertyOverride
+        is_set: true
```
Comment (shuyama1) [inline mmv1/products/compute/terraform.yaml:1965]: `is_set` is probably not needed here as it's already defined as map

File: mmv1/third_party/terraform/tests/resource_compute_per_instance_config_test.go [hunk 5]
```diff
@@ -125,6 +125,47 @@ func TestAccComputePerInstanceConfig_update(t *testing.T) {
 	})
 }
 
+func TestAccComputePerInstanceConfig_statefulIps(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (hao-nan-li) [reply mmv1/third_party/terraform/tests/resource_compute_per_instance_config_test.go:128]: Thanks for pointing out, the update test is supposed to have a different value in `auto_delete` for the stateful Ips, let me change that.

File: mmv1/products/compute/terraform.yaml (hunk 4 above)
Comment (hao-nan-li) [reply mmv1/products/compute/terraform.yaml:1965]: Sure I will take it out.

File: mmv1/products/compute/api.yaml (hunk 3 above)
Comment (hao-nan-li) [reply mmv1/products/compute/api.yaml:9757]: I don't see a clean description on the GCP public doc, perhaps the current description is good enough.

**Review by shuyama1 — APPROVED** [review]
Would you mind rebasing your PR?

**Comment by shuyama1** [comment]
@hao-nan-li failing test is unrelated. Feel free to merge

**Pull request merged**